lib/routes/account.js

```diff
@@ -1530,6 +1590,80 @@ module.exports = function (
     },
     {
       method: 'POST',
+      path: '/account/login/send_unblock_code',
+      config: {
+        validate: {
+          payload: {
+            email: validators.email().required()
+          }
+        }
+      },
+      handler: function (request, reply) {
+        log.begin('Account.SendUnblockCode', request)
+
+        var email = request.payload.email
+        var ip = request.app.clientAddress
+        var emailRecord
+
+        return customs.check(request, email, 'sendUnblockCode')
+          .then(lookupAccount)
+          .then(createUnblockCode)
+          .then(mailUnblockCode)
+          .done(function () {
+            reply({})
+          }, reply)
+
+        function lookupAccount() {
+          return db.emailRecord(email)
+            .then(function(record) {
+              emailRecord = record
+              return record.uid
+            })
+        }
+
+        function createUnblockCode(uid) {
+          return db.createUnblockCode(uid)
+        }
+
+        function mailUnblockCode(code) {
+          return getGeoData(ip)
+            .then(function (geoData) {
+              return mailer.sendUnblockCode(emailRecord, code, userAgent.call({
+                acceptLanguage: request.app.acceptLanguage,
+                ip: ip,
+                location: geoData.location,
+                timeZone: geoData.timeZone
+              }, request.headers['user-agent'], log))
+            })
+        }
+      }
+    },
+    {
+      method: 'POST',
+      path: '/account/login/reject_unblock_code',
+      config: {
+        validate: {
+          payload: {
+            uid: isA.string().max(32).regex(HEX_STRING).required(),
+            unblockCode: isA.string(BASE_36).max(6).required()
+          }
+        }
+      },
+      handler: function (request, reply) {
+        var uid = Buffer(request.payload.uid, 'hex')
+        var code = request.payload.unblockCode
+
+        log.begin('Account.RejectUnblockCode', request)
+        db.consumeUnblockCode(uid, code)
+          .then(
+            function () {
+              return {}
```

Comment:
I'm super excited about having this available for metrics purposes.   An `activityEvent` is probably not what we want (e.g. doing an reject_unblock_code should't count as activity for DAU purposes) but even a `log.info` will be valuable to start.